src/webots/core/WbProject.cpp

```diff
@@ -75,14 +71,38 @@ WbProject *WbProject::defaultProject() {
   return gDefaultProject;
 }
 
-WbProject *WbProject::extraDefaultProject() {
-  if (gExtraDefaultProject == NULL && !WbPreferences::instance()->value("General/extraProjectsPath").toString().isEmpty() &&
-      QDir(WbPreferences::instance()->value("General/extraProjectsPath").toString() + "/default/").exists()) {
-    gExtraDefaultProject =
-      new WbProject(WbPreferences::instance()->value("General/extraProjectsPath").toString() + "/default/");
-    qAddPostRoutine(WbProject::cleanupExtraDefaultProject);
+QList<WbProject *> *WbProject::extraProjects() {
+  if (gExtraProjects == NULL) {
+    gExtraProjects = new QList<WbProject *>();
+    // collect extra project paths in a QSet to avoid duplicate entries
+    QSet<QString> projectPaths;
+
+    if (!WbPreferences::instance()->value("General/extraProjectsPath").toString().isEmpty()) {
+      foreach (const QString &path, WbPreferences::instance()
+                                      ->value("General/extraProjectsPath")
+                                      .toString()
+                                      .split(QDir::listSeparator(), Qt::SkipEmptyParts))
```

Comment:
```suggestion
    if (!WbPreferences::instance()->value("General/extraProjectPath").toString().isEmpty()) {
      foreach (const QString &path, WbPreferences::instance()
                                      ->value("General/extraProjectPath")
                                      .toString()
                                      .split(QDir::listSeparator(), Qt::SkipEmptyParts))
```

Reply:
I've discussed internally and we decided on keeping it singular, like with the environmental variable `WEBOTS_EXTRA_PROJECT_PATH`, no "s".
So we should probably maintain consistency by keeping it singular everywhere.
(might need to do a wide search in case it's referenced elsewhere other than WbPreferences)

Reply:
It is referenced in WbPreferencesDialog and, correct me if I'm wrong, the translations should be modified too.
This could be a breaking change for somebody migrating from 2022a, is there any extra documentation you want me to add for this?

Reply:
There are contributors that sometimes take it upon themselves to do the translations for other languages, but most of them are incomplete, and in practice, rarely used. As for ourselves, we only aim to have the English version up to date as we don't have the resource nor knowledge to ensure all others are properly translated. By doing changes to the english strings, the others are flagged as needing adjustment on its own.

I don't think this change breaks backwards compatibility as it extends functionality without removing anything that was there before, so we're fine on that front (unless I missed something, are you referring to something in particular?)
Mentioning the added feature in the changelog as you did is sufficient.

Reply:
Ah I see what you meant for the translations, then yes it needs to be updated as well.

For the backwards compatibility you're right, it does technically require some adjustment by the user so it should be mentioned in bold in the changelog. You can see an example of this [here](https://cyberbotics.com/doc/reference/changelog-r2021). It should be sufficient to notify the user that if they had any extra project paths they should re-set the field in the preferences.

Reply:
Changed naming, translation files, and changelog